test/extensions/filters/http/lua/lua_filter_test.cc

```diff
@@ -1992,6 +1992,35 @@ TEST_F(LuaHttpFilterTest, InspectStreamInfoDowstreamSslConnectionOnPlainConnecti
   EXPECT_EQ(Http::FilterHeadersStatus::Continue, filter_->decodeHeaders(request_headers, true));
 }
 
+// Should survive from multiple streamInfo():downstreamSslConnection() calls.
+// This is a regression test for #14091.
+TEST_F(LuaHttpFilterTest, SurviveMultipleDownstreamSslConnectionCalls) {
+  const std::string SCRIPT{R"EOF(
+    function envoy_on_request(request_handle)
+      if request_handle:streamInfo():downstreamSslConnection() ~= nil then
+         request_handle:logTrace("downstreamSslConnection is present")
+      end
+    end
+  )EOF"};
+
+  setup(SCRIPT);
+
+  auto connection_info = std::make_shared<Ssl::MockConnectionInfo>();
```

Comment:
Nit. use `const`? And since you're here probably the above one too.

Reply:
Thanks for a review. I will fix it. Just to be sure - by "above one" you mean the other instance of `auto connection_info = std::make_shared<Ssl::MockConnectionInfo>();` in this file?

Reply:
Yes, you're correct. Thank you so much!

Reply:
done